src/bootstrap.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+# This script allows auto-bootstrapping by downloading a suitable D compiler.
+# Usage: ./bootstrap.sh <build-args>
+# It is a wrapper around src/build.d and all arguments will be passed to build.d
+# For more information, please see the documentation of build.d.
+#
+# We recommend installing a D compiler globally and using src/build.d directly.
+# Visit https://dlang.org/download.html for more information
+
+HOST_DMD_VER="${HOST_DMD_VER:-2.088.0}"
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
```

Comment:
Why use multiple subshells instead of something like this?

```suggestion
DIR="$( dirname "$PWD/${BASH_SOURCE[0] )"
```

Reply:
There are edge cases with e.g. symlinks.
A full discussion: https://stackoverflow.com/questions/59895/how-to-get-the-source-directory-of-a-bash-script-from-within-the-script-itself

Reply:
Good to know.